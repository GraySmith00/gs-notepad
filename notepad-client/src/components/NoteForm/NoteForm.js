import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './NoteForm.scss';

class NoteForm extends Component {
  state = {
    text: '',
    tag: 'work',
    error: ''
  };

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = e => {
    e.preventDefault();
    this.props.addNote(this.state);
    this.setState({ text: '', tag: 'work' });
  };

  render() {
    const { text, tag } = this.state;
    const { postError } = this.props;

    return (
      <React.Fragment>
        <form onSubmit={this.handleSubmit} className="note-form">
          <textarea
            onChange={this.handleChange}
            type="text"
            name="text"
            value={text}
            className="text-input"
            placeholder="New note..."
            rows="3"
          />
          <select
            onChange={this.handleChange}
            name="tag"
            value={tag}
            className="tag-select"
          >
            <option value="work">Work</option>
            <option value="personal">Personal</option>
            <option value="hobby">Hobby</option>
          </select>

          <button>SAVE</button>
        </form>
        {postError && <p className="error-text">{postError}</p>}
      </React.Fragment>
    );
  }
}

NoteForm.propTypes = {
  addNote: PropTypes.func.isRequired
};

export default NoteForm;
