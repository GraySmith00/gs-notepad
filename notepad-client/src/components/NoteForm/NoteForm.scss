.note-form {
  display: flex;
  align-items: flex-start;
  margin-bottom: 1.5rem;

  textarea {
    border: none;
    border-radius: 5px;
    width: 40%;
    padding: 1.5rem;
    font-size: 1.6rem;
  }

  select {
    height: 3rem;
    margin-left: 1.5rem;
  }

  button {
    height: 3rem;
    padding: 0 2rem;
    margin-left: 1.5rem;
    border-radius: 5px;
    color: #fff;
    background: #cb2d3e; /* fallback for old browsers */
    background: -webkit-linear-gradient(
      to right,
      #ef473a,
      #cb2d3e
    ); /* Chrome 10-25, Safari 5.1-6 */
    background: linear-gradient(
      to right,
      #ef473a,
      #cb2d3e
    ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  }
}
