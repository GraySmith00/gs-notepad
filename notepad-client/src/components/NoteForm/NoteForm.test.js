import React from 'react';
import { shallow } from 'enzyme';

import NoteForm from './NoteForm';

describe('NoteForm component', () => {
  let wrapper;
  let mockAddNote;

  beforeEach(() => {
    mockAddNote = jest.fn();
    wrapper = shallow(<NoteForm addNote={mockAddNote} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change the state on input change', () => {
    expect(wrapper.find('.text-input').length).toEqual(1);

    wrapper
      .find('.text-input')
      .simulate('change', { target: { name: 'text', value: 'get groceries' } });

    expect(wrapper.state().text).toEqual('get groceries');
  });

  it('should change the state on select change', () => {
    expect(wrapper.find('.tag-select').length).toEqual(1);

    wrapper
      .find('.tag-select')
      .simulate('change', { target: { name: 'tag', value: 'hobby' } });

    expect(wrapper.state().tag).toEqual('hobby');
  });

  it('should call addNote on submit', () => {
    expect(wrapper.find('.note-form').length).toEqual(1);

    wrapper
      .find('.note-form')
      .simulate('submit', { preventDefault: jest.fn() });

    expect(mockAddNote).toHaveBeenCalled();
  });

  it('should reset the state', () => {
    wrapper.setState({ text: 'hi', tag: 'work' });

    wrapper
      .find('.note-form')
      .simulate('submit', { preventDefault: jest.fn() });

    expect(wrapper.state().text).toEqual('');
    expect(wrapper.state().tag).toEqual('work');
  });
});
