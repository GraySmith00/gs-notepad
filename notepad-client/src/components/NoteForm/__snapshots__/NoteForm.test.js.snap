// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteForm component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoteForm
    addNote={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="text"
        >
          Text
          <input
            className="text-input"
            name="text"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>,
        <label>
          Tag
          <select
            className="tag-select"
            name="tag"
            onChange={[Function]}
            value="work"
          >
            <option
              value="work"
            >
              Work
            </option>
            <option
              value="personal"
            >
              Personal
            </option>
            <option
              value="hobby"
            >
              Hobby
            </option>
          </select>
        </label>,
        <button>
          SAVE
        </button>,
      ],
      "className": "note-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Text",
            <input
              className="text-input"
              name="text"
              onChange={[Function]}
              type="text"
              value=""
            />,
          ],
          "htmlFor": "text",
        },
        "ref": null,
        "rendered": Array [
          "Text",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "text-input",
              "name": "text",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Tag",
            <select
              className="tag-select"
              name="tag"
              onChange={[Function]}
              value="work"
            >
              <option
                value="work"
              >
                Work
              </option>
              <option
                value="personal"
              >
                Personal
              </option>
              <option
                value="hobby"
              >
                Hobby
              </option>
            </select>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Tag",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="work"
                >
                  Work
                </option>,
                <option
                  value="personal"
                >
                  Personal
                </option>,
                <option
                  value="hobby"
                >
                  Hobby
                </option>,
              ],
              "className": "tag-select",
              "name": "tag",
              "onChange": [Function],
              "value": "work",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Work",
                  "value": "work",
                },
                "ref": null,
                "rendered": "Work",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Personal",
                  "value": "personal",
                },
                "ref": null,
                "rendered": "Personal",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Hobby",
                  "value": "hobby",
                },
                "ref": null,
                "rendered": "Hobby",
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SAVE",
        },
        "ref": null,
        "rendered": "SAVE",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="text"
          >
            Text
            <input
              className="text-input"
              name="text"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>,
          <label>
            Tag
            <select
              className="tag-select"
              name="tag"
              onChange={[Function]}
              value="work"
            >
              <option
                value="work"
              >
                Work
              </option>
              <option
                value="personal"
              >
                Personal
              </option>
              <option
                value="hobby"
              >
                Hobby
              </option>
            </select>
          </label>,
          <button>
            SAVE
          </button>,
        ],
        "className": "note-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Text",
              <input
                className="text-input"
                name="text"
                onChange={[Function]}
                type="text"
                value=""
              />,
            ],
            "htmlFor": "text",
          },
          "ref": null,
          "rendered": Array [
            "Text",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "text-input",
                "name": "text",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Tag",
              <select
                className="tag-select"
                name="tag"
                onChange={[Function]}
                value="work"
              >
                <option
                  value="work"
                >
                  Work
                </option>
                <option
                  value="personal"
                >
                  Personal
                </option>
                <option
                  value="hobby"
                >
                  Hobby
                </option>
              </select>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Tag",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="work"
                  >
                    Work
                  </option>,
                  <option
                    value="personal"
                  >
                    Personal
                  </option>,
                  <option
                    value="hobby"
                  >
                    Hobby
                  </option>,
                ],
                "className": "tag-select",
                "name": "tag",
                "onChange": [Function],
                "value": "work",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Work",
                    "value": "work",
                  },
                  "ref": null,
                  "rendered": "Work",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Personal",
                    "value": "personal",
                  },
                  "ref": null,
                  "rendered": "Personal",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Hobby",
                    "value": "hobby",
                  },
                  "ref": null,
                  "rendered": "Hobby",
                  "type": "option",
                },
              ],
              "type": "select",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SAVE",
          },
          "ref": null,
          "rendered": "SAVE",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
